PeerioTechnologies/peerio-desktop
# chore: replace RTB components Link, Card, Checkbox, FontIcon

#### Relevant info and issue/PR links 
https://app.clubhouse.io/peerio/story/1272/desktop-replace-card-and-link-components
https://app.clubhouse.io/peerio/story/1273/desktop-replace-checkbox-components
https://app.clubhouse.io/peerio/story/1285/desktop-replace-fonticon-component

#### Testing instructions  
This is one of those annoying `chore`s where a lot of things are affected but in a really minor way.

1. All `Card` and `Link` RTB components have been replaced with plain markup. There were only a few no need to test as they're extremely simple and I've verified that they look/work as they should.
2. All `Checkbox` RTB components have been replaced with our own `Checkbox` component. There was one `Button` that was pretending to be a checkbox that has also been converted. Here are their locations:
  * 2FA login dialog
  * logout "untrust device" dialog
  * URL preview "only from favourites"
  * "Share from my files" dialog (multi-select files)
3. All `FontIcon` RTB components have been replaced with our own `MaterialIcon`. This is every instance of an icon that is *not* in a `Button`, `ListItem`, `MenuItem`... Basically this affects any instance of a Material icon that is used purely decoratively and is not clickable/interactable. While these are kind of everywhere through the app, I don't think it needs to be tested intensively, due to the lack of interactivity. I gave each page a once-over when I updated it so IMO it's fine.

----
### Repository owner


[PR guideline](https://github.com/PeerioTechnologies/peerio-icebear/blob/dev/docs/CONTRIBUTING.md)
